tests/unit/http/test_http_client.py

```diff
@@ -134,6 +134,20 @@ def test_last_response_empty_on_error(self):
             self.assertIsNotNone(self.client.last_request)
             self.assertIsNone(self.client.last_response)
 
+    def test_request_with_proxy(self):
+        self.client.proxy = {
+            'http': 'http://proxy.twilio.com:8118',
+            'https': 'https://proxy.twilio.com:8118',
+        }
+        self.request_mock.url = 'https://api.twilio.com/'
+        self.request_mock.headers = {'Host': 'other.twilio.com'}
+
+        response = self.client.request(
+            'doesnt matter', 'doesnt matter', None, None, None, None, None, None)
+
+        self.assertEqual('other.twilio.com', self.request_mock.headers['Host'])
+        self.assertEqual(200, response.status_code)
```

Comment:
This test isn't really testing anything; just that setting the proxy doesn't break anything.

Better to test that the proxy gets applied to the session:
```
self.assertEqual(self.client.proxy, self.session_mock.proxies)
```

Reply:
updated.